consensus/src/main/kotlin/maru/consensus/validation/BlockValidator.kt

```diff
@@ -111,10 +110,11 @@ class TimestampValidator(
 
 class ProposerValidator(
   private val proposerSelector: ProposerSelector,
+  private val beaconChain: BeaconChain,
 ) : BlockValidator {
   override fun validateBlock(block: BeaconBlock): SafeFuture<Result<Unit, BlockValidationError>> =
     proposerSelector
-      .getProposerForBlock(block.beaconBlockHeader.toConsensusRoundIdentifier())
+      .getProposerForBlock(beaconChain.getLatestBeaconState(), block.beaconBlockHeader.toConsensusRoundIdentifier())
```

Comment:
Instead of using the latest state, this should use the state for the parent block, like we are doing in the `ProposerSelectorAdapter`. This way we this code is repeatable and we can validate older blocks if necessary.

Reply:
Correct, I fixed it